demo/site/src/blocks/InternalLinkBlock.tsx

```diff
@@ -0,0 +1,30 @@
+"use client";
+import { PropsWithData } from "@comet/cms-site";
+import { InternalLinkBlockData } from "@src/blocks.generated";
+import Link from "next/link";
+import * as React from "react";
+
+interface InternalLinkBlockProps extends PropsWithData<InternalLinkBlockData> {
+    children: React.ReactNode;
+    title?: string;
+}
+
+export function InternalLinkBlock({ data: { targetPage, targetPageAnchor }, children, title }: InternalLinkBlockProps): React.ReactElement {
+    if (!targetPage) {
+        return <>{children}</>;
+    }
+
+    let href = targetPageAnchor !== undefined ? `${targetPage.path}#${targetPageAnchor}` : targetPage.path;
+    if (targetPage.scope) {
+        const language = (targetPage.scope as Record<string, string>).language;
+        if (language) {
+            href = `/${language}${href}`;
+        }
+    }
```

Comment:
Yes, a factory would be an alternative. That basePath is dynamic, that makes all this pretty complicated.

In our last meeting we talked about this and also had the factory as idea but decided to copy. Because it's not that much code to copy after all.